### PR TITLE
feat(sessions): Add option to filter by proj slug in query params [INGEST-1210]

### PR DESCRIPTION
**Pull request by iker-barriocanal**
Dashboards require the ability to filter by projects in query parameters, and right now this is only possible by the top-level `project` parameter. This PR introduces the feature to filter project slug and environment in the query. Filtering by the top-level parameter and filtering int he query at the same time is also supported.

Filtering by project slug is supported by adding `project_id` to the condition column list. Filtering by environment is a bit different - no need to explicitly add anything, it's enough to look for conditions in nested lists in the conditions (i.e. check conditions even if they are in `[[[<condition>]]]` format, instead of only considering `[<condition>]`).

### CODE REVIEW

File: src/sentry/snuba/sessions_v2.py [hunk 1]
```diff
@@ -324,15 +324,49 @@ def __init__(
...
         filter_keys = {
             resolve_filter_key(key): value for key, value in snuba_filter.filter_keys.items()
         }
 
         self.aggregations = snuba_filter.aggregations
-        self.conditions = conditions
         self.filter_keys = filter_keys
 
+    def _get_conditions(self, sn_filter_conditions, col_resolver, con_cols, max_nested_depth=5):
+        # Conditions in nested lists with depth >max_nested_depth are ignored.
```
Comment (iker-barriocanal) [inline src/sentry/snuba/sessions_v2.py:340]: The max depth I've seen here is 2, but I wanted to make sure we don't make too many calls here.

File: src/sentry/snuba/sessions_v2.py [hunk 2]
```diff
@@ -324,15 +324,49 @@ def __init__(
...
+        def helper(cons, depth):
+            if depth <= 0:
+                return []
```
Comment (iker-barriocanal) [inline src/sentry/snuba/sessions_v2.py:343]: This piece here silently drops conditions, to not return error responses. We may want to consider raising an error, or logging something with Sentry.

File: tests/snuba/sessions/test_sessions_v2.py [hunk 3]
```diff
@@ -222,6 +224,51 @@ def test_virtual_groupby_query():
     assert query.query_groupby == []
 
 
+class SnubaSessionsV2Test(TestCase, SnubaTestCase):
+    # This test suite needs a project to exist in the database.
```
Comment (iker-barriocanal) [inline tests/snuba/sessions/test_sessions_v2.py:228]: I considered a better alternative to create this class over mocking plenty of things. Easier to read and maintain, imo.

File: src/sentry/snuba/sessions_v2.py [hunk 4]
```diff
@@ -324,15 +324,49 @@ def __init__(
...
+
         filter_keys = {
             resolve_filter_key(key): value for key, value in snuba_filter.filter_keys.items()
         }
 
         self.aggregations = snuba_filter.aggregations
-        self.conditions = conditions
         self.filter_keys = filter_keys
 
+    def _get_conditions(self, sn_filter_conditions, col_resolver, con_cols, max_nested_depth=5):
```
Comment (untitaker) [inline src/sentry/snuba/sessions_v2.py:339]: can't this function be put into resolve_tags?

File: src/sentry/snuba/sessions_v2.py (hunk 1 above)
Comment (untitaker) [reply src/sentry/snuba/sessions_v2.py:340]: I would go for unbounded recursion here tbh

**Comment by iker-barriocanal** [comment]
Closing in favor of https://github.com/getsentry/sentry/pull/34281.

**Pull request closed without merging**